lib/utils/device/deploy.ts

```diff
@@ -244,10 +247,11 @@ export async function deployToDevice(opts: DeviceDeployOptions): Promise<void> {
 	// Now that we've set the target state, the device will do it's thing
 	// so we can either just display the logs, or start a livepush session
 	// (whilst also display logs)
+	const promises: Array<Promise<void>> = [streamDeviceLogs(api, opts)];
```

Comment:
This refactoring (across multiple commits) achieves 3 things:
- Reduces code duplication - there were 2 copies of `getLoStream`, `displayDeviceLogs`, `outputDeferredMessages`..., now there is only one copy.
- Simplifies the logic (isolates some code in a separate function, easier to read). 
- Adds `livepush.close()` to a `finally` block to allow the CLI process to exit if the connection to the device is lost the requested number of connection retries has been reached.